File: mydagger2/src/main/java/com/zhongjh/mydagger2/entity/MainActivity.java
package com.zhongjh.mydagger2.entity;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.zhongjh.mydagger2.R;

import javax.inject.Inject;

/**
 * @author zhongjh
 * @date 2022/4/24
 */
public class MainActivity extends AppCompatActivity {

    @Inject
    MainEntity mainEntity;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerMainComponent.builder().build().inject(this);
        mainEntity.setName("名称");
        mainEntity.getMainParameter().setName("名称2");


        ((TextView) findViewById(R.id.tvName)).append(mainEntity.getName() + "_" + mainEntity.getMainParameter().getName());
    }


}

File: myhilt/src/main/java/com/zhongjh/hiltdemo/noarguments/MainActivity.kt
package com.zhongjh.hiltdemo.noarguments

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.zhongjh.hiltdemo.R
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @set:Inject
    lateinit var mainEntity: MainEntity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        mainEntity.name = "名称"

//        tvName.append(mainEntity.name)
    }

}